packages/synapse-react-client/src/components/SynapseTable/TopLevelControls/TopLevelControls.tsx

```diff
@@ -274,55 +212,54 @@ const TopLevelControls = (props: TopLevelControlsProps) => {
               <Divider orientation="vertical" variant="middle" flexItem />
             </>
           )}
-          {controls.map(control => {
-            const { key, icon, tooltipText } = control
-            if (
-              (key === 'showDownloadConfirmation' && hideDownload) ||
-              (key === 'showFacetVisualization' && hideVisualizationsControl) ||
-              (key === 'showSqlEditor' && hideSqlEditorControl) ||
-              (key === 'showCopyToClipboard' && !showCopyToClipboard)
-            ) {
-              // needs to be a file view in order for download to make sense
-              return <React.Fragment key={key}></React.Fragment>
```

Comment:
This was also confusing IMO. Better to colocate the "hide" logic with the control instead of with "hide" conditions for other controls.